test/integration/backward_compatible/parallel/map/MapLockTest.js

```diff
@@ -87,44 +86,36 @@ describe('MapLockTest', function () {
         await testFactory.shutdownAll();
     });
 
-    it('should acquire the lock when key owner terminates', function (done) {
-        let clientTwo;
-        let keyOwner;
-        let key;
-
-        RC.startMember(cluster.id).then((m) => {
-            keyOwner = m;
-            return testFactory.newHazelcastClientForParallelTests({
+    it('should acquire the lock when key owner terminates', async function () {
+        let done = false;
+        const keyOwner = await RC.startMember(cluster.id);
+        const clientTwo = await testFactory.newHazelcastClientForParallelTests({
                 clusterName: cluster.id,
                 properties: {
                     ['hazelcast.client.invocation.timeout.millis']: INVOCATION_TIMEOUT_FOR_TWO
                 }
-            }, [member, m]);
-        }).then((c) => {
-            clientTwo = c;
-            return waitForNewPartitionTable(client, keyOwner);
-        }).then(() => {
-            key = generateKeyOwnedBy(client, keyOwner);
-            return map.lock(key);
-        }).then(() => {
-            return clientTwo.getMap('test');
-        }).then((mapOnTwo) => {
-            // try to lock concurrently
-            mapOnTwo.lock(key)
-                .then(() => {
-                    return mapOnTwo.unlock(key);
-                })
-                .then(() => clientTwo.shutdown())
-                .then(done)
-                .catch(done);
-            return TestUtil.promiseWaitMilliseconds(2 * INVOCATION_TIMEOUT_FOR_TWO);
+        }, [member, keyOwner]);
+        await waitForNewPartitionTable(client, keyOwner);
+        const key = generateKeyOwnedBy(client, keyOwner);
+        await map.lock(key);
+        const mapOnTwo = await clientTwo.getMap('test');
+        // try to lock concurrently
+        mapOnTwo.lock(key).then(() => {
+            return mapOnTwo.unlock(key);
         }).then(() => {
-            return map.isLocked(key);
-        }).then((locked) => {
-            expect(locked).to.be.true;
-            return RC.terminateMember(cluster.id, keyOwner.uuid);
+            clientTwo.shutdown();
```

Comment:
In the original version of the test, we were waiting for this call as well. so, we can add the missing return here

Reply:
You are right, good catch 👍🏻